core/jvm/src/test/scala/scalaz/zio/RTSSpec.scala

```diff
@@ -150,20 +150,20 @@ class RTSSpec(implicit ee: ExecutionEnv) extends Specification with AroundTimeou
     ) must_=== Some(ExampleError)
 
   def testEvalOfAttemptOfFail = Seq(
-    unsafeRun(IO.fail[Throwable, Int](ExampleError).attempt[Throwable]) must_=== Left(ExampleError),
-    unsafeRun(IO.suspend(IO.suspend(IO.fail[Throwable, Int](ExampleError)).attempt[Throwable])) must_=== Left(
+    unsafeRun(IO.fail[Throwable](ExampleError).attempt) must_=== Left(ExampleError),
+    unsafeRun(IO.suspend(IO.suspend(IO.fail[Throwable](ExampleError)).attempt)) must_=== Left(
       ExampleError
     )
   )
 
   def testAttemptOfDeepSyncEffectError =
-    unsafeRun(deepErrorEffect(100).attempt[Throwable]) must_=== Left(ExampleError)
+    unsafeRun(deepErrorEffect(100).attempt) must_=== Left(ExampleError)
 
   def testAttemptOfDeepFailError =
-    unsafeRun(deepErrorFail(100).attempt[Throwable]) must_=== Left(ExampleError)
+    unsafeRun(deepErrorFail(100).attempt) must_=== Left(ExampleError)
 
   def testEvalOfUncaughtFail =
-    unsafeRun(IO.fail[Throwable, Int](ExampleError)) must (throwA(UnhandledError(ExampleError)))
+    unsafeRun(IO.fail[Throwable](ExampleError)).asInstanceOf[Any] must (throwA(UnhandledError(ExampleError)))
```

Comment:
To eliminate these casts, how about we add some method to `IO` for narrowing the value:

```scala
def as[A1](implicit ev: A <~< A1): IO[E, A1] = self.asInstanceOf[IO[E, A1]]
```

Then we can do:

```scala
unsafeRun(IO.fail[Throwable](ExampleError).as[Any]) must ...
```

And eliminate all the `asInstanceOf` coercions.

Reply:
Nice, I will do that